google/resource_compute_instance_iam_test.go

```diff
@@ -0,0 +1,306 @@
+
+	project := getTestProjectFromEnv()
+	account := acctest.RandomWithPrefix("tf-test")
+	role := "roles/compute.osLogin"
+	region := getTestRegionFromEnv()
+	zone := getTestZoneFromEnv()
+	subnetwork := fmt.Sprintf("tf-test-net-%s", acctest.RandString(10))
+	instanceName := fmt.Sprintf("tf-test-instance-%s", acctest.RandString(10))
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				// Test Iam Member creation (no update for member, no need to test)
+				Config: testAccComputeInstanceIamMember_basic(project, account, region, zone, subnetwork, instanceName, role),
+			},
+			{
+				ResourceName:      "google_compute_instance_iam_member.foo",
+				ImportStateId:     fmt.Sprintf("%s/%s/%s %s serviceAccount:%s@%s.iam.gserviceaccount.com", project, zone, instanceName, role, account, project),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccComputeInstanceIamPolicy(t *testing.T) {
+	t.Parallel()
+
+	project := getTestProjectFromEnv()
+	account := acctest.RandomWithPrefix("tf-test")
+	role := "roles/compute.osLogin"
+	region := getTestRegionFromEnv()
+	zone := getTestZoneFromEnv()
+	instanceName := fmt.Sprintf("tf-test-instance-%s", acctest.RandString(10))
+	subnetwork := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeInstanceIamPolicy_basic(project, account, region, zone, subnetwork, instanceName, role),
+			},
+			// Test a few import formats
+			{
+				ResourceName:      "google_compute_instance_iam_policy.foo",
+				ImportStateId:     fmt.Sprintf("projects/%s/zones/%s/instances/%s", project, zone, instanceName),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				ResourceName:      "google_compute_instance_iam_policy.foo",
+				ImportStateId:     fmt.Sprintf("%s/%s/%s", project, zone, instanceName),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				ResourceName:      "google_compute_instance_iam_policy.foo",
+				ImportStateId:     fmt.Sprintf("%s/%s", zone, instanceName),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccComputeInstanceIamMember_basic(project, account, region, zone, subnetworkName, instanceName, roleId string) string {
+	return fmt.Sprintf(`
+resource "google_service_account" "test_account" {
+  account_id   = "%s"
+  display_name = "Iam Testing Account"
+}
+
+resource "google_compute_network" "network" {
+  name = "%s"
+  auto_create_subnetworks = false
+}
+
+resource "google_compute_subnetwork" "subnetwork" {
+  name = "%s"
+  region = "%s"
+  ip_cidr_range = "10.1.0.0/16"
+  network = "${google_compute_network.network.name}"
+}
+
+resource "google_compute_instance" "test_vm" {
+	project = "%s"
```

Comment:
Looks like there is a mix of tabs and spaces in the config. Can you standardize on one?

Reply:
Done!